pyrogram/client/ext/dispatcher.py

```diff
@@ -188,8 +188,12 @@ async def update_worker(self, lock):
 
                             if isinstance(handler, handler_type):
                                 try:
-                                    if handler.check(parsed_update):
-                                        args = (parsed_update,)
+                                    if callable(handler.filters) and asyncio.iscoroutinefunction(handler.filters.__call__):
```

Comment:
`callable(handler.filters)` is not actually needed. Filter objects in the current API always have `__call__` defined, if they don't they are not proper filters and pyrogram will raise an exception when accessing `handler.filters.__call__`, which is better than having it silently ignore updates.

Reply:
I think we should remove all of these checks here and have `handler.check` actually check for coroutine functions. It also helps removing one repeated line `args = (parsed_update,)`.

Reply:
sure.